pydm/widgets/embedded_display.py

```diff
@@ -142,12 +137,28 @@ def open_file(self):
         display : QWidget
         """
         # Expand user (~ or ~user) and environment variables.
-        fname = os.path.expanduser(os.path.expandvars(self.filename))
-        if os.path.isabs(fname):
-            return self.app.open_file(fname, macros=self.parsed_macros())
-        else:
-            return self.app.open_relative(fname, self,
-                                          macros=self.parsed_macros())
+        if not is_pydm_app():
+            return
+        if not self._needs_load:
```

Comment:
Seems odd to me that this won't return the widget we've previously loaded from the file at least? What is my recourse here if I think the file has changed and want to reload?